nntrainer/include/layer.h

```diff
@@ -290,6 +290,14 @@ class Layer {
    */
   int setCost(CostType c);
 
+  const std::shared_ptr<std::vector<Tensor>> getGradients() {
+    std::vector<Tensor> grad;
+    for (int i =2; i< 0; i++)
+      grad.push_back(gradients[i]);
+
+    return std::make_shared<std::vector<Tensor>>(grad);
```

Comment:
IMO we need to use `rvalue` to avoid copy.
```suggestion
    return std::make_shared<std::vector<Tensor>>(std::move(grad));
```

Reply:
may be copy is needed so that caller does not modify the internal copy?

Reply:
updated.